### ISSUE TITLE
Test Accuracy

### ISSUE THREAD

**Issue by MDS198** [JiaruQian/GHOST#2]
I cloned the repository and ran the evaluation on Cora and Citeseer datasets **without any modifications** to the code or configurations. The test accuracy I obtained is much lower than the results reported in the paper. Additionally, I noticed the accuracy on Citeseer is higher than on Cora, which seems unusual based on typical benchmark behavior.

- Command: `python main.py --dataset cora`
  - Test Loss: 1.6123
  - Test Accuracy: **0.3795**
- Command: `python main.py --dataset citeseer`
  - Test Loss: 1.2314
  - Test Accuracy: **0.4132**

### Questions
1. Could you confirm the expected accuracy for these datasets with the current codebase?
2. Are there any specific environment requirements (e.g., specific PyTorch Geometric version) that might cause this drop in accuracy?
3. Is it expected that Citeseer accuracy exceeds Cora accuracy in this implementation? (Usually Cora is higher.)

Thank you for your time and for providing this code!

**Comment by JiaruQian** [comment]
Hi there,Thank you for your interest in our work and for reaching out. I will address your questions point by point:

1.  Reproducibility and Performance Deviations

The experimental results reported in our paper are reproducible using the current codebase. However, performance drops (or improvements) can occur due to inconsistencies in the software environment or hardware configurations.
(a) Environment: To ensure consistency, please verify that your environment matches our core package versions:
torch_cluster: 1.6.3+pt23cu121
pyg_lib: 0.4.0+pt23cu121
torch_scatter: 2.1.2+pt23cu121
torch_sparse: 0.6.18+pt23cu121
torch: 2.3.1 (with CUDA 12.1)
(b) Hardware Platform: Our experiments were conducted using NVIDIA GeForce RTX 3090 GPUs and Intel(R) Xeon(R) CPU E5-2678 v3 @ 2.50GHz. (The CPU model matters here because the calculation of $L_{fgw}$ in our paper is offloaded to the CPU). Different hardware architectures handle floating-point arithmetic logic differently, which can lead to cumulative numerical errors and impact the final accuracy.


2. Accuracy Trends (Cora vs. Citeseer)

Under standard benchmarks, Cora typically yields higher accuracy than Citeseer, which is also reflected in our paper's reported results. However, our experimental setting involves extreme data distribution heterogeneity (non-IID). In this scenario, the volume of data held by different clients varies significantly. Under such high-variance distributions, it is possible to observe atypical behavior where Citeseer results occasionally exceed Cora.

In addition, we have updated a more detailed version of run.sh ( with the setting of hyperparameter of each dataset). Let us know if the issue persists!

**Comment by MDS198** [comment]
Thank you for your reply! I have the following two more questions regarding the implementation details:

**1. Evaluation Logic in `utils/taskflow.py`**
I noticed this snippet:

```python
server_nodes = []
for data in clients_data:
    for node in range(len(data.global_map)):
        if data.test_mask[node]:
            server_nodes.append(data.global_map[node])
```
It seems the evaluation strictly filters nodes based on test_mask without including their training/validation neighbors in the subgraph. In traditional transductive settings, neighbors are usually included (but not evaluated).

**2. Discrepancy in Results Between Windows and Linux**
I ran the exact same Python script (same seed) on Windows and Linux, but there was a significant difference in the final accuracy/ratings. Could you please let me know whether the code was primarily developed/tested on Windows or Linux?

**Comment by JiaruQian** [comment]
Hi there.

1. This is a very insightful observation. Thank you for pointing this out and for your interest in the implementation details. Our current implementation's logic was primarily driven by the privacy constraints between clients in the FGL context—we aimed to minimize data leakage and ensure strict isolation during the evaluation phase.
However, we agree that this part of the evaluation flow could be more flexible to better align with traditional GNN benchmarks. We appreciate your feedback and will look into optimizing this logic in future updates to better support different subgraph sampling strategies while maintaining privacy.

2. Linux.

Thank you again for helping us improve the project!
